api/views/course_admin.py

```diff
@@ -16,11 +17,13 @@ class CourseAdminViewSet(viewsets.GenericViewSet):
     permission_classes = [TeacherAccessPermission, ]
     queryset = CanvasCourse.objects.all()
     serializer_class = CourseSerializer
+    search_fields = ['user__first_name', 'user__last_name', 'user__email']
```

Comment:
There is a logical problem here. Notice that the view set is for the model `CanvasCourse` and has a `CourseSerializer`. Now you added search fields for `user__first_name` and ... which are properties of `CanvasCourseRegistration` and not `CanvasCourse`. Then you are manually calling the filter for a `CanvasCourseRegistration` queryset. That is why it works for that specific action. But it will break the general functionality of the view set.